guides/development-workflow.md

```diff
@@ -0,0 +1,341 @@
+
+When working in a team other pull requests will often get merged in before yours is finished  It's important to make sure you're developing against the latest version of the code base as much as possible.
+
+You can do this by rebasing your work on top of the `develop` branch.
+
+```
+git checkout <branch-name>
+git rebase origin/develop
+git push --force-with-lease
+```
+
+During the rebase there may be conflicts which should be resolved at this point. If in doubt talk to the other developer who you're conflicting with and reason out what the correct state should be.
+
+Forcing pushing in this situation is the only time it should be necessary, adding `--force-with-lease` ensures you don't accidentally overwrite someone else's additions.
+
+## Committing
+
+Writing a good commit message serves three important purposes.
+
+* to speed up the review process
+* to encourage individual commits to be well small, bounded and concise
+* to help future maintainers (it could be you) in say 3 years time, to find out why a particular change was made and what context was relevant at the time
+
+**Convention:**
+
+```
+(feature): when we are adding something new
+(fix): when we are fixing something existing
+(refactor): when we are changing the implementation of something existing without changing its behaviour
+(chore): catch all for when none of the above apply and there is no immediate user need eg. gem/package updates
+```
+
+**Examples:**
+
+```
+# Poor
+Added repairs
+
+# Great
+(feature): Job postings are now displayed on the landing page
+
+* Getting the Job postings out of the external API proved difficult but I think the solution we have will allow us to easily upgrade to the new release they have planned: <link>
+```
+
+```
+# Poor
+Fixed the form
+
+# Great
+(fix): Food order form works with special characters
+
+* Previously the form failed to send if the starter value included any of the following characters: `!@£$%^&`
+* Users should be able to use these characters and after a short investigation I found this new bug in PHP: <link>
+* As a short term workaround until the bug is fixed I am stripping out any of these characters if they are present.
+```
+
+```
+# Poor
+Gem upgrade
+
+# Great
+(chore): Upgrade widget to v1.2.1
+
+* This widget now includes a patch that improves resilience: <link>
+* We are still unable to upgrade to v2.x as we need to do some migration work first
+```
+---
+
+## Code review
+
+## Code style
+We write our code in a consistent way to ensure it is well-structured and easy to follow for the rest of the team. Similar to the guidelines laid out in [Thoughtbot’s style guide](https://playbook.thoughtbot.com/#style-guide), approach this guidance as a way to code on present and future projects, not something to retroactively add to existing projects. When working on existing projects or joining a project, make sure to follow the code style that is already in place.
```

Comment:
This was section was moved from the `playbook.md` without my modification however my opinion on this is that by not following this by default, codebases can easily become like spaghetti with umpteen implementations of the same thing, it becomes harder to read. 

I'm not for a moment suggesting that we shouldn't improve the code, just that this should be the result of a conscious decision made by the team to ensure we all pull in the same direction from A to B.

It's important developers don't join a project and start introducing new styles so a project is left with A, B, C, D, E, F and a team that doesn't have any idea which to use.